client/src/util/exponentialBackoffFetch.js

```diff
@@ -0,0 +1,68 @@
+export default function createBackoffFetch(retryTimes = 3) {
+  let prevHash = null;
+  let retryCount = 0;
+  let response = null;
+  let error = null;
+  let promise = null;
+
+  // This will only be called by itself or when we've seen a new matrix
+  const backoffFetch = (fetchArgs) => {
+    const promiseExecuter = (resolve, reject) => {
+      retryCount += 1;
+      // We don't want to timeout our first request
+      if (retryCount === 1) {
```

Comment:
is there a way to remove a conditional block for `retryCount === 1` and streamline the exponential backoff?

Something like: 

setTimeout(() => {}, retryCount * 700 * 5 ** retryCount)

So we get:
`retryCount === 0: 0`
`retryCount === 1: 3500`
`retryCount === 2: 35000`
`retryCount === 3: 262500`

Just trying to see if we could streamline the code XD Thank you!

Reply:
`setTimeout( ___, 0)` will still timeout the function until the call stack empties, didn't want to add a perf dip in immediately successful requests

Reply:
https://flaviocopes.com/javascript-event-loop/

Reply:
ah I was thinking right after `backoffFetch(fetchArgs)` is synchronously executed, the `setTimeout(() => {}, 0)` callback function will be the next task in the stack, so there shouldn't be any other synchronous calls in the stack?

<img width="1624" alt="Screen Shot 2020-12-13 at 11 36 29 PM" src="https://user-images.githubusercontent.com/6309723/102053246-42369d00-3d9c-11eb-8825-470410e2ae48.png">

Reply:
Just seeing this now.  I might be wrong but isn't the status of the call stack at the time which the timeout added to the stack pretty indeterminate? We have a few ticks going off in the background like manual GC, WebGL re-renders and other user interactions.  I ended up pulling the timeout cb into its own function and then just using it in multiple places, that way its still performant while keeping it readable and DRY